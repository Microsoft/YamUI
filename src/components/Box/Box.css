/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
@import "src/css/variables/colors.css";

.y-box {
  position: relative;
  padding: 8px;
  border-radius: 2px;
  border: 1px solid $color__indieRock;
}
