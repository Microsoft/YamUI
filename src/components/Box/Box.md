### Notes for use

A Box component can be used with any number of children and all of them will be inside the box.
Use it in cases where you would consider using a Block but need a border.

### Examples

Basic usage:

```js { "props": { "data-example": "basic" } }
<Box>This is a Box</Box>
```

Box with MediaObject child:

```js { "props": { "data-example": "with-mediaobject-child" } }
const { AvatarSize } = require('../Avatar');
const { MediaObjectSize } = require('../MediaObject');
const avatar = (size) => (
  <Avatar imageUrl={user.imageUrl} name={user.name} size={size} />
);

<Box>
  <MediaObject
    size={MediaObjectSize.XLARGE}
    imageContent={avatar(AvatarSize.XLARGE)}
    titleContent={user.name}
    metadataContent={user.jobTitle}
    extraContent={user.status}
  />
</Box>
```
