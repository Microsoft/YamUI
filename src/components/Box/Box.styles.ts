/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { semanticColors } from '../../util/colors';
import { mergeStyleSets } from '@uifabric/styling';
import { memoizeFunction } from '@uifabric/utilities';

export interface BoxStyleProps {
  hasOnClick: boolean;
}

export const getClassNames = memoizeFunction((styleProps: BoxStyleProps) => {
  const { hasOnClick } = styleProps;

  return mergeStyleSets({
    root: {
      borderRadius: '1px',
      border: `1px solid ${semanticColors.bodyDivider}`,
      cursor: hasOnClick ? 'pointer' : undefined,
    },
    inner: {
      margin: '-1px',
      position: 'relative',
    },
  });
});
