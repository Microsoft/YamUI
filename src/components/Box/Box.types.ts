/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import { PaletteColor } from '../../util/colors';

export interface BoxProps extends NestableBaseComponentProps {
  /**
   * Called when box is clicked.
   */
  onClick?: (() => void);

  /**
   * Background color of the box.
   */
  backgroundColor?: PaletteColor;

  /**
   * Color of the border.
   */
  borderColor?: PaletteColor;
}
