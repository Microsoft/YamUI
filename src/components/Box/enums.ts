/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum BoxBorderType {
  ROUND = 'round',
  SOFT = 'soft',
  SQUARE = 'square',
}

export enum BoxPaddingSize {
  LARGE = 'large',  // 16px
  MEDIUM = 'medium', // 12px
  SMALL = 'small', // 8px
}
