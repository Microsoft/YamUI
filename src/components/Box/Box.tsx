/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import {
  NestableBaseComponentProps as BoxProps,
} from '../../util/BaseComponent/props';
import Block, { GutterSize } from '../Block';
import './Box.css';

export { BoxProps };
/**
 * A Block component with a border. Usually used as a container component
 */
export default class Box extends React.Component<BoxProps, {}> {
  render() {
    const { children } = this.props;

    return (
      <Block className={this.getClasses()} padding={GutterSize.SMALL}>
        {children}
      </Block>
    );
  }

  private getClasses() {
    const { className } = this.props;
    const classes = ['y-box'];

    if (className) {
      classes.push(className);
    }

    return classes.join(' ');
  }
}
