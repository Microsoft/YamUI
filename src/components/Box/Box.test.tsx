/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Box, { BoxProps } from './index';

describe('<Box />', () => {
  let component: ShallowWrapper<BoxProps, {}>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<Box />);
    });

    it('has its correct base class', () => {
      expect(component.hasClass('y-box')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(
        <Box className="TEST_CLASSNAME" />,
      );
    });

    it('includes that className', () => {
      expect(component.hasClass('TEST_CLASSNAME')).toBe(true);
    });

    it('still includes its base className', () => {
      expect(component.hasClass('y-box')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with children passed', () => {
    beforeEach(() => {
      component = shallow(<Box/>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
