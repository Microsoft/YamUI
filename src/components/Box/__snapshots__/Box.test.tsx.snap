// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> with additional className matches its snapshot 1`] = `
<Block
  className="y-box TEST_CLASSNAME"
  padding="small"
/>
`;

exports[`<Box /> with children passed matches its snapshot 1`] = `
<Block
  className="y-box"
  padding="small"
/>
`;

exports[`<Box /> with default options matches its snapshot 1`] = `
<Block
  className="y-box"
  padding="small"
/>
`;
